### ISSUE TITLE
binary addition fails

### ISSUE THREAD

**Issue by gchiu** [metaeducation/ren-c#571]
>> a: copy #{00}
== #{00}

>> b: copy #{01}
== #{01}

>> add a b
** Script Error: invalid argument: #{01}
** Where: add
** Near: add a b ??
** File: ../src/core/t-string.c
** Line: 1049

**Comment by kealist** [comment]
I don't think that adding `binary!` makes sense.  It may seem to in this particular case, but it is complicated in most situations: `#{00} + #{11112132}`.   That being said, I do wish there were better tools in rebol for dealing with binary.

**Comment by gchiu** [comment]
>> add-binary: func [ a b][a: to integer! a b: to integer! b to binary! a + b]
== make function! [[a b return:][
    return: make function! [
        ["Returns a value from a function." value [<opt> any-value!]]
        [exit/from/with (context-of 'return) :value]
    ]
    (a: to integer! a b: to integer! b to binary! a + b)
]]

>> add-binary #{01} #{02}
== #{0000000000000003}

>> add-binary #{00} #{11112132}
== #{0000000011112132}

>> add-binary #{01} #{FF}
== #{0000000000000000}

>> add-binary #{00} #{FF}
== #{FFFFFFFFFFFFFFFF}

**Comment by kealist** [comment]
Is this a feature request?

**Comment by gchiu** [comment]
Yes.  We can add binary! and decimal! together at present.

>> add copy #{FE}  1
== #{FF}

and I'm working with a binary! for controlling lights etc so I need to increase values easily
